test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -219,6 +223,155 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
 	return nil
 }
 
+// checkPodExists returns true if the pod exists in the given namespace.
+func (data *testData) checkPodExists(podName, namespace string) (bool, error) {
+	_, err := data.e2eTestData.PodWaitFor(1*time.Second, podName, namespace, nil)
+	return err == nil, nil
+}
+
+// getPodPortsAndIPs retrieves the interfaces and IPs of a target Pod.
+func (data *testData) getPodPortsAndIPs(targetPod *testPodInfo) (map[string][]string, error) {
+	cmd := []string{"ip", "addr", "show"}
+	stdout, _, err := data.e2eTestData.RunCommandFromPod(data.e2eTestData.GetTestNamespace(), targetPod.podName, containerName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("error listing interfaces for Pod %s: %w", targetPod.podName, err)
+	}
+
+	portsAndIPs := make(map[string][]string)
+	var ifaceName string
+
+	for _, line := range strings.Split(stdout, "\n") {
+		fields := strings.Fields(line)
+		if len(fields) >= 2 && strings.HasSuffix(fields[0], ":") {
+			ifaceName = strings.Split(strings.TrimSuffix(fields[1], ":"), "@")[0]
+		}
+		if len(fields) > 1 && (fields[0] == "inet" || fields[0] == "inet6") {
+			portsAndIPs[ifaceName] = append(portsAndIPs[ifaceName], fields[1])
+		}
+	}
+	return portsAndIPs, nil
+}
+
+// getNetAttachDefClient returns a NetAttachDef client.
+func (data *testData) getNetAttachDefClient() (netattdef.Interface, error) {
+	config, err := data.e2eTestData.GetKubeConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kubeconfig: %w", err)
+	}
+	return netattdef.NewForConfig(config)
+}
+
+// getIPPoolNames retrieves the IPPool names from a NetworkAttachmentDefinition.
+func (data *testData) getIPPoolNames(networkName, namespace string) ([]string, error) {
+	client, err := data.getNetAttachDefClient()
+	if err != nil {
+		return nil, err
+	}
+
+	nad, err := client.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get NetworkAttachmentDefinition %s: %w", networkName, err)
+	}
+
+	var nadConfig struct {
+		IPAM struct {
+			IPPools []string `json:"ippools"`
+		} `json:"ipam"`
+	}
+
+	if err := json.Unmarshal([]byte(nad.Spec.Config), &nadConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse NAD config JSON: %w", err)
+	}
+
+	return nadConfig.IPAM.IPPools, nil
+}
+
+// checkIPReleased verifies if the VLAN IP is released.
+func (data *testData) checkIPReleased(ipPoolName, podIPString string) error {
+	crdClient, err := data.e2eTestData.GetCRDClient()
+	if err != nil {
+		return fmt.Errorf("failed to get CRD client: %w", err)
+	}
+
+	timeout := time.After(40 * time.Second)
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
```

Comment:
we always use the polling functions for this (e.g., `wait.PollUntilContextTimeout`)

Reply:
done